### PR TITLE
Document standalone Drools server

### LINKED ISSUE #386: Document connecting to a separate rules server
This work began in #274, but I think we need an addition to INSTALL.md that fully describes how to connect to the rules server and import the PSM's rules into it.  If not in INSTALL, then maybe in a separate doc in the `docs` directory (linked from INSTALL).

Once #385 is handled satisfactorily, we'll likely suggest to devs that they set up and connect to a separate rules server to have a dev setup that is closer to demo (and eventually prod) configurations.

### PR DESCRIPTION
**Pull request by jasonaowen**
Document the steps needed to set up a standalone Drools server, load our rules into it, and configure the PSM to talk to it.

Resolves #386 Document connecting to a separate rules server

### CODE REVIEW

File: psm-app/build.gradle [hunk 1]
```diff
@@ -165,6 +165,8 @@ project(':cms-web') {
 project(':cms-business-model') {
     apply plugin: 'com.github.jacobono.jaxb'
     apply plugin: 'java'
+    sourceCompatibility = 1.6
+    targetCompatibility = 1.6
```
Comment (cecilia-donnelly) [inline psm-app/build.gradle]: Does including this in `build.gradle` pose a problem for our local builds?

File: docs/Drools UI.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+application requires JBoss 7.1.1. The Drools installer has automated that
+setup, so that to get a running instance needs only a few steps:
+
+1. [Download Drools](https://downloads.sourceforge.net/project/jbpm/jBPM%205/jbpm-5.4.0.Final/jbpm-5.4.0.Final-installer-full.zip)
+2. Unzip
+3. Run installer: `ant install.demo.noeclipse`
+4. Start server: `ant start.demo.noeclipse`
+
+If you are on a RedHat / CentOS machine, [we've written a script you can
+use](../scripts/drools-microservice.sh).
```
Comment (cecilia-donnelly) [inline docs/Drools UI.md]: nice. :)

File: docs/Drools UI.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+In order for the Drools server to be able to understand the rules, it needs to
+have a copy of the relevant classes.
+
+1. Build them with `./gradlew cms-business-model:build`, which produces
+   `cms-business-model/build/libs/cms-business-model.jar`.
+2. Upload this jar to the Drools server
+    1. Create a new package named `cms`
+    2. Upload `cms-business-model.jar`: 'Create New' > 'Upload POJO Model jar'
+        1. Name it 'cms-business-model'
+        2. Create in package `cms`
```
Comment (cecilia-donnelly) [inline docs/Drools UI.md]: These are all instructions in the GUI on the Drools server, right?

File: docs/Drools UI.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
++                   type='DRL'
++                   basicAuthentication="enabled"
++                   username="username"
++                   password="password"
++         />
+```
+
+The default username and password the jBPM installer configures is
+`admin`/`admin`. Substitute your URL, username, and password in the three
+rules configuration XML files.
```
Comment (cecilia-donnelly) [inline docs/Drools UI.md]: Were the username and password created in the Drools installer?

File: docs/Drools UI.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+enrollment. Form validation is performed by Drools, so trying to advance from
+an empty enrollment page should give you several "this field is required"
+errors. If you get a server error, examine the server log for more details as
+to what went wrong.
+
+You can also add a new rule to the Drools server. Currently, the PSM loads the
+rules the first time it needs to use Drools for anything (effectively, the
+first time someone starts an enrollment), so you will need to reload the
+application to pick up the new rules. One way to do that is to run
+`jboss-cli.sh --connect --command=reload`.
```
Comment (cecilia-donnelly) [inline docs/Drools UI.md]: Nice, thanks for this!

File: psm-app/build.gradle (hunk 1 above)
Comment (jasonaowen) [reply psm-app/build.gradle]: No, it shouldn't! More recent versions of Java can both compile for and run code compiled for older versions, and we're not using any Java 7 or Java 8 features in that subproject.

File: docs/Drools UI.md (hunk 2 above)
Comment (jasonaowen) [reply docs/Drools UI.md]: Yeah, good job @slifty! :)

File: docs/Drools UI.md (hunk 3 above)
Comment (jasonaowen) [reply docs/Drools UI.md]: Right. Do you have any suggestions on making that more clear?

File: docs/Drools UI.md (hunk 4 above)
Comment (jasonaowen) [reply docs/Drools UI.md]: Yes, but I don't think the installer provides an opportunity to change them - you'd need to use the JBoss management console or `jboss-cli.sh` to do so.

**Pull request merged**